ncm-opennebula/src/main/perl/OpenNebula/AII.pm

```diff
@@ -0,0 +1,433 @@
+
+Detect and process C<OpenNebula> C<VM> templates.
+
+=cut
+
+sub process_template_aii
+{
+    my ($self, $config, $tt_name, $oneversion) = @_;
+
+    my $tree = $config->getElement('/')->getTree();
+    if ((defined $oneversion) and ($oneversion >= version->new("5.0.0"))) {
+        $tree->{system}->{opennebula}->{boot} = $BOOT_V5;
+        $self->verbose("BOOT section set to support OpenNebula versions >= 5.0.0");
+    } else {
+        $self->verbose("BOOT section set to support OpenNebula versions < 5.0.0");
+        $tree->{system}->{opennebula}->{boot} = $BOOT_V4;
+    };
+
+    # TBD process_template refactoring
+    my $tpl = CAF::TextRender->new(
+        $tt_name,
+        $tree,
+        relpath => 'opennebula',
+        log => $self,
+        );
+    if (!$tpl) {
+        $self->error("TT processing of $tt_name failed.", $tpl->{fail});
+        return;
+    }
+    return "$tpl";
+}
+
+=item read_one_aii_conf
+
+Reads a config file in C<.ini> style with a minimal RPC endpoint setup.
+Returns an C<OpenNebula> instance afterwards.
+
+=cut
+
+sub read_one_aii_conf
+{
+    my ($self, $data) = @_;
+
+    my $rpc = "rpc";
+
+    if (! -f $AII_OPENNEBULA_CONFIG) {
+        $self->error("No configfile $AII_OPENNEBULA_CONFIG.");
+        return;
+    }
+
+    my $config = Config::Tiny->new;
+    my $domainname = $data->getElement ($DOMAINNAME)->getValue;
+    my $hostname = $data->getElement ($HOSTNAME)->getValue;
+    my $fqdn = "${hostname}.${domainname}";
+
+    $config = Config::Tiny->read($AII_OPENNEBULA_CONFIG);
+    foreach my $section (sort keys %{$config}) {
+        $self->verbose("Found RPC section: $section");
+        my $pattern = $config->{$section}->{pattern};
+        if ($pattern and $fqdn =~ /^$pattern$/ and $rpc eq 'rpc') {
+            $rpc = $section;
+            $self->info("Match pattern in RPC section: [$rpc]");
+            last;
+        };
+    };
+    if (exists($config->{$domainname}) and $rpc eq 'rpc') {
+        $rpc = $domainname;
+        $self->info ("Detected configfile RPC section: [$rpc]");
+    };
+    $config->{$rpc}->{port} //= $ONE_DEFAULT_PORT;
+    my $port = $config->{$rpc}->{port};
+    my $host = $config->{$rpc}->{host};
+    $config->{$rpc}->{url} //= $ONE_DEFAULT_URL;
+    $config->{$rpc}->{user} //= $ONEADMIN_USER;
+
+    # Keep backwards compatibility
+    if ($host) {
+        $self->warn("RPC old host section detected: $host. ",
+                              "Please use metaconfig to generate a proper OpenNebula aii configuration.",
+                              "ONE aii will replace the RPC url by the assigned host at this point.");
+        $config->{$rpc}->{url} = "http://$host:$port/RPC2";
+    };
+
+    if (! $config->{$rpc}->{password} ) {
+        $self->error("No password set in configfile $AII_OPENNEBULA_CONFIG. Section [$rpc]");
+        return;
+    };
+
+    my $one = $self->make_one($config->{$rpc});
+    return $one;
+
+}
+
+=item is_supported_one_version
```

Comment:
what does this have to do with AII? maybe make a `NCM::Components::OpenNebula::Common` and put these generic methods in there?